File: src/main/java/com/example/api/models/Usuario.java
package com.example.api.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "usuarios")
@Data
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Idusuario")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "idPersona")
    private Persona persona;

    @Column(name = "Correo")
    private String correo;

    @Column(name = "Contraseña")
    private String contrasenia;

    @ManyToOne
    @JoinColumn(name = "id_rol")
    private Rol rol;

    private String estado;

}

File: WebApplication1/Models/Usuario.cs
namespace WebApplication1.Models
{
    public class Usuario
    {
        public long id { get; set; }
        public Persona persona { get; set; }
        public string correo { get; set; }
        public string contrasenia { get; set; }
        public Rol rol { get; set; }
        public string estado { get; set; }
    }
}
